### ISSUE TITLE
[BUG] Names don't adhere to plugin naming convention

### ISSUE THREAD

**Issue by dblock** [opensearch-project/opensearch-plugins#174]
### What is the bug?

We have published a plugins naming convention to https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#plugin-naming-conventions in #56, but are failing to adhere to these consistently 

- [index-management-dashboards-plugin](https://github.com/opensearch-project/index-management-dashboards-plugin)
- [observability](https://github.com/opensearch-project/observability)
- [security-analytics-dashboards-plugin](https://github.com/opensearch-project/security-analytics-dashboards-plugin) 
- [security-dashboards-plugin](https://github.com/opensearch-project/security-dashboards-plugin)
- [dashboards-maps](https://github.com/opensearch-project/dashboards-maps)
- [dashboards-visualizations](https://github.com/opensearch-project/dashboards-visualizations)
...

### What is the expected behavior?

Evaluate the delta and align all names to the convention.

**Comment by elfisher** [comment]
@mnkugler could you help pick this one up?

**Comment by mnkugler** [comment]
@elfisher sure!

**Comment by mnkugler** [comment]
Can anyone comment on the implications of renaming a plugin? For example, how do we handle existing dependencies? With an alias?

**Comment by davidlago** [comment]
May the first brave team renaming their plugin serve as a tracer bullet for this 🎖️ 

If this ends up being bigger than a breadbox, we should have the discussion around whether the juice is worth the squeeze especially for plugin repos given that we are thinking of what the next iteration of plugins would be and they might be on a (albeit long) deprecation path. 🤔

**Comment by macohen** [comment]
Oh man. Renaming plugins sounds like a painful process. But, can we ensure that naming is consistent moving forward? In what cases from the [CONVENTIONS.md](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#plugin-naming-conventions) doc does "These are recommended plugin naming conventions. YMMV." really apply if we do want consistency?

**Comment by dblock** [comment]
+1 on getting rid of YMMV

**Comment by mnkugler** [comment]
Here's a summary of the most noticeable deviations from the conventions. 

| Plugin | Issue (links to the rule in CONVENTIONS.md) |
|---|---|
| [Index Management Dashboard](https://github.com/opensearch-project/index-management-dashboards-plugin) | [Remove plugin from the repo name](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-plugins) |
| [Alerting Dashboard](https://github.com/opensearch-project/alerting-dashboards-plugin) | [Remove plugin from the repo name](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-plugins) |
| [Security Analytics Dashboard Plugins](https://github.com/opensearch-project/security-analytics-dashboards-plugin) | [Remove plugin from the repo name](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-plugins) |
| [Security Dashboard Plugin](https://github.com/opensearch-project/security-dashboards-plugin) | [Remove plugin from the repo name](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-plugins) |
| [Dashboard Maps](https://github.com/opensearch-project/dashboards-maps) | [Folder needs to be named appropriately (i.e. custom-import-map) with no underscores](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Dashboard Maps](https://github.com/opensearch-project/dashboards-maps) | [Plugin ID needs to be lowercase camelcase](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-dashboard-plugins-1) |
| [Dashboard Maps](https://github.com/opensearch-project/dashboards-maps) | [Meta needs to be updated to match the plugin name](https://github.com/opensearch-project/opensearch-plugins/blob/main/dashboards-plugins/.meta) |
| [Dashboard Visualization](https://github.com/opensearch-project/dashboards-visualizations) | [Folder needs to Include full plugin name in kebab case](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Alerting](https://github.com/opensearch-project/alerting) | [Folder needs to Include full plugin name in kebab case, e.g., plugins/opensearch-alerting](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [SQL](https://github.com/opensearch-project/sql) | [Folder needs to Include full plugin name in kebab case, e.g., plugins/opensearch-sql](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Anomaly Detection](https://github.com/opensearch-project/anomaly-detection) | [Folder needs to be created and Include full plugin name in kebab case, e.g., plugins/opensearch-anomaly-detection](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Asynchronous Search](https://github.com/opensearch-project/asynchronous-search) | [Folder needs to be created and Include full plugin name in kebab case, e.g., plugins/opensearch-asynchronous-search](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Common Utils](https://github.com/opensearch-project/common-utils) | [Folder needs to be created and Include full plugin name in kebab case, e.g., plugins/opensearch-common-utils](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Common Utils](https://github.com/opensearch-project/common-utils) | [opensearchplugin metadata is missing from build.gradle](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#metadata) |
| [Dashboard Reports](https://github.com/opensearch-project/dashboards-reports/tree/main) | [Folder needs to be created and Include full plugin name in kebab case, e.g., plugins/notebooks-dashboards](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Index Management ](https://github.com/opensearch-project/index-management) | [Folder needs to be created and Include full plugin name in kebab case, e.g., plugins/opensearch-index-management](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#folders) |
| [Performance Analyzer RCA](https://github.com/opensearch-project/performance-analyzer-rca) | [opensearchplugin metadata is missing from build.gradle](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#metadata) |
| [ML Commons](https://github.com/opensearch-project/ml-commons) | [opensearchplugin metadata is missing from build.gradle](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#metadata) |
| [Geospatial](https://github.com/opensearch-project/geospatial) | [opensearchplugin metadata is missing from build.gradle](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#metadata) |
| [i18n](https://github.com/opensearch-project/i18n-plugin) | [Remove plugin from the repo name](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-plugins) |
| [i18n](https://github.com/opensearch-project/i18n-plugin) | [Plugin ID needs to align with convention with camel case, e.g., i18nDashboards](https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md#opensearch-dashboard-plugins-1) |

**Comment by amitgalitz** [comment]
For current plugins that have already been published to maven without following the convention, will we create a new plugin name and release under a new name?

**Comment by dblock** [comment]
@amitgalitz Do you have a strong preference one way or another? Feels like 3.0 may be a good time to do this.

**Comment by amitgalitz** [comment]
> @amitgalitz Do you have a strong preference one way or another? Feels like 3.0 may be a good time to do this.

If we are thinking long term, I think 3.0 is a good time to do this as well. Developers should be expecting breaking changes, having to potentially change API, or gradle dependencies during 3.0 bump could make sense. 

There are 2 things I can think of at least that we should consider before though. 
1. We should find the correct way to let users know of the change to a new release on maven since I don't know of an easy way to link one maven software to a separate maven software and label that as a continuation. 

2. We should only actually change plugin names for convention case if we figure out a way to really enforce these naming conventions. Even as recent as 2.4.0 we have new plugins that used the new onboarding issue creation process that still have incorrect plugin names.

**Comment by mnkugler** [comment]
Dashboard Maps, SQL, Common Util, & Performance Analyzer RCA have all resolved their issues. The following table illustrates the 10 Repos left and their status: 
**Repo**|**Issue** |**Owner**|**Status**|**GH Issue**
-----|-----|-----|-----|-----
Index Management Dashboard|Remove plugin from the repo name|getsaurabh02|will start to fix on April 24th|[issue](https://github.com/opensearch-project/index-management-dashboards-plugin/issues/415)
Alerting Dashboard|Remove plugin from the repo name|getsaurabh02|Fix by 3.0|[issue](https://github.com/opensearch-project/alerting-dashboards-plugin/issues/372)
Security Analytics Dashboard Plugins|Remove plugin from the repo name|praveensameneni|TBD|[issue](https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/187)
Security Dashboard Plugin|Remove plugin from the repo name|davidlago|Fix by 3.0|[issue](https://github.com/opensearch-project/security-dashboards-plugin/issues/1232)
Dashboard Visualization|Folder needs to be fixed|anirudha|On hold until 2.7 and Ani will come up with a solution|[issue](https://github.com/opensearch-project/dashboards-visualizations/issues/130#issuecomment-1503960022)
Alerting|Folder needs to be fixed|getsaurabh02|Waiting on Alerting to onboard with job scheduler to resolve|[issue](https://github.com/opensearch-project/alerting/issues/687)
ML Commons|opensearchplugin metadata is missing from build.gradle|sean-zheng-amazon|TBD|[issue](https://github.com/opensearch-project/ml-commons/issues/847)
Geospatial|opensearchplugin metadata is missing from build.gradle|vamshin|TBD|[issue](https://github.com/opensearch-project/geospatial/issues/252)
i18n|Remove plugin from the repo name & Plugin ID needs to align with convention with camel case|seanneumann|Fix currently in progress|[issue](https://github.com/opensearch-project/dashboards-i18n/pull/22)
Asynchronous Search|Folder needs to be fixed|dhruvdas|TBD|[issue](https://github.com/opensearch-project/asynchronous-search/issues/209)

**Comment by mnkugler** [comment]
Updated chart: Alerting, Index Management, and Security Analytics is planning to work on removing the word “plugin” from the repo name after 2.11 to ensure the best timing for no breaking changes.
